File: Reference-Android-Java/app/src/main/java/xyz/fay/reference/feature/weather/WeatherViewModel.java
package xyz.fay.reference.feature.weather;

import android.content.Context;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import xyz.fay.reference.networking.NetworkManager;
import xyz.fay.reference.networking.RequestHandler;
import xyz.fay.reference.networking.response.WeatherResponse;

public class WeatherViewModel extends ViewModel {
    private final MutableLiveData<WeatherResponse> weatherResponse = new MutableLiveData<>();

    public void fetchData(Context context) {
        NetworkManager manager = new NetworkManager();
        manager.getWeather(context, new RequestHandler<WeatherResponse>() {
            @Override
            public void completion(WeatherResponse response) {
                weatherResponse.setValue(response);
            }
        });
//        manager.getWeather(context, response -> {
//            weatherResponse.setValue(response);
//        });
//        manager.getWeather(context, weatherResponse::setValue);
    }

    public MutableLiveData<WeatherResponse> getWeatherResponse() {
        return weatherResponse;
    }
}

File: Reference-Android-Kotlin/app/src/main/java/xyz/fay/reference/feature/weather/WeatherViewModel.kt
package xyz.fay.reference.feature.weather

import android.content.Context
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import xyz.fay.reference.networking.NetworkManager
import xyz.fay.reference.networking.response.WeatherResponse

class WeatherViewModel : ViewModel() {
    private val _weatherResponse = MutableLiveData<WeatherResponse>()
    val weatherResponse: MutableLiveData<WeatherResponse>
        get() = _weatherResponse

    fun fetchData(context: Context) {
        val manager = NetworkManager()
        manager.getWeather(context) {
            _weatherResponse.value = it
        }
//        manager.getWeather(context, _weatherResponse::setValue)
    }
}
